src/components/BackdropModal.js

```diff
@@ -0,0 +1,318 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useRef } from 'react';
+import {
+  Modal,
+  View,
+  StyleSheet,
+  TouchableWithoutFeedback,
+  Animated,
+  Dimensions,
+  PanResponder,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import { COLORS } from '../styles/themes';
+
+const { height: screenHeight } = Dimensions.get('window');
+
+/**
+ * A custom modal component that extends React Native's Modal with:
+ * - Fade backdrop animation (never slides)
+ * - Configurable content animation (slide, fade, none)
+ * - Swipe-to-dismiss functionality
+ * - Touch outside to dismiss
+ * - Flexible positioning and styling
+ */
+const BackdropModal = ({
+  visible,
+  onDismiss,
+  children,
+  animationType = 'slide',
+  position = 'bottom',
+  enableSwipeToDismiss = true,
+  enableBackdropPress = true,
+  backdropColor = 'rgba(0, 0, 0, 0.5)',
+  contentStyle,
+  containerStyle,
+  swipeThreshold = 100,
+  onShow,
+  onHide,
+  ...modalProps
+}) => {
+  const backdropOpacity = useRef(new Animated.Value(0)).current;
+  const slideAnim = useRef(new Animated.Value(screenHeight)).current;
+  const gestureAnim = useRef(new Animated.Value(0)).current;
+  const scaleAnim = useRef(new Animated.Value(0.8)).current;
+
+  // Pan responder for swipe to dismiss
+  const panResponder = useRef(
+    PanResponder.create({
+      onStartShouldSetPanResponder: () => enableSwipeToDismiss,
+      onMoveShouldSetPanResponder: (evt, gestureState) => {
+        if (!enableSwipeToDismiss) return false;
+
+        // Only respond to vertical swipes
+        const { dy, dx } = gestureState;
+        return Math.abs(dy) > Math.abs(dx) && Math.abs(dy) > 10;
+      },
+      onPanResponderMove: (evt, gestureState) => {
+        if (gestureState.dy > 0) {
+          gestureAnim.setValue(gestureState.dy);
+        }
+      },
+      onPanResponderRelease: (evt, gestureState) => {
+        if (gestureState.dy > swipeThreshold || gestureState.vy > 0.5) {
+          handleDismiss();
+        } else {
+          // Snap back to original position
+          Animated.spring(gestureAnim, {
+            toValue: 0,
+            useNativeDriver: true,
+          }).start();
+        }
+      },
+    })
+  ).current;
+
+  const handleDismiss = () => {
+    if (!onDismiss) return;
+
+    // Animate out
+    const animations = [
+      Animated.timing(backdropOpacity, {
+        toValue: 0,
+        duration: 300,
+        useNativeDriver: false,
+      }),
+    ];
+
+    if (animationType === 'slide') {
+      animations.push(
+        Animated.timing(slideAnim, {
+          toValue: screenHeight,
```

Comment:
Is this sliding the modal to this height? What does it mean? I though the `toValue` would be 0 `onDismiss`. Maybe add some comments